### PR TITLE
feat(file-router): add menu items generation

### PR DESCRIPTION
**Pull request by Lodin**

### CODE REVIEW

**Comment by taefi** [comment]
Is there an issue that describes the expectations for this PR?

**Review by taefi — CHANGES_REQUESTED** [review]

File: packages/java/endpoint/src/main/java/com/vaadin/hilla/route/ClientRouteRegistry.java [hunk 1]
```diff
@@ -40,8 +40,8 @@ public class ClientRouteRegistry implements Serializable {
      *
      * @return a list of all registered routes
      */
-    public List<ClientViewConfig> getAllRoutes() {
-        return List.copyOf(registeredRoutes.values());
+    public List<Map.Entry<String, ClientViewConfig>> getAllRoutes() {
```
[the remark this answers is not in the document]
Comment (Lodin) [reply packages/java/endpoint/src/main/java/com/vaadin/hilla/route/ClientRouteRegistry.java]: Yeah, I didn't think enough on this structure. Today I noticed that issue and decided to replace it with a regular `Map` (copy of internal `LinkedHashMap`).

File: packages/ts/hilla-file-router/package.json [hunk 2]
```diff
@@ -47,27 +50,32 @@
...
+    "chai-as-promised": "^7.1.1",
     "chai-like": "^1.1.1",
     "deep-equal-in-any-order": "^2.0.6",
     "mocha": "^10.2.0",
     "rimraf": "^5.0.5",
-    "sinon": "^17.0.1",
+    "sinon": "^16.0.0",
     "type-fest": "^4.9.0",
-    "typescript": "^5.3.3"
+    "typescript": "5.3.2"
```
Comment (platosha) [inline packages/ts/hilla-file-router/package.json:78]: nit: since the plugin uses TS API, `typescript` is a runtime dependency

Comment (Lodin) [reply packages/ts/hilla-file-router/package.json:78]: Right, thanks

**Review by platosha — APPROVED** [review]

**Review by taefi — APPROVED** [review]

**Review by platosha — APPROVED** [review]

**Pull request merged**